src/opensearchDatasource.ts

```diff
@@ -662,6 +676,100 @@ export class OpenSearchDatasource
     );
   }
 
+  async getLiveStreamKey(query: OpenSearchQuery): Promise<string> {
+    const str = JSON.stringify({ query: query.query });
+    const msgUint8 = new TextEncoder().encode(str); // encode as (utf-8) Uint8Array
+    const hashBuffer = await crypto.subtle.digest('SHA-1', msgUint8); // hash the message
+    const hashArray = Array.from(new Uint8Array(hashBuffer.slice(0, 8))); // first 8 bytes
+    return `${query.datasource?.uid}/${hashArray.map((b) => b.toString(16).padStart(2, '0')).join('')}/${config.bootData.user.orgId}`;
+  }
+
+  runLiveQueryThroughBackend(request: DataQueryRequest<OpenSearchQuery>): Observable<DataQueryResponse> {
```

Comment:
We could also probably extract this function to a separate file to keep the datasource class cleaner